### PR TITLE
[BOLT] [Profile] Fix type mismatch error

### PR DESCRIPTION
**Pull request by gz83**
issue: https://github.com/llvm/llvm-project/issues/73006

@maksfb

### CODE REVIEW

File: bolt/lib/Profile/YAMLProfileReader.cpp [hunk 1]
```diff
@@ -354,7 +357,15 @@ Error YAMLProfileReader::readProfile(BinaryContext &BC) {
       matchProfileToFunction(YamlBF, Function);
   }
 
-  for (auto &[CommonName, LTOProfiles]: LTOCommonNameMap) {
```
Comment (maksfb) [inline bolt/lib/Profile/YAMLProfileReader.cpp:357]: Could you please try to build with:
```
    for (const auto &[CommonName, LTOProfiles]: LTOCommonNameMap) {
```
? I don't have a Windows builder available to test.

File: bolt/lib/Profile/YAMLProfileReader.cpp [hunk 2]
```diff
@@ -354,7 +357,15 @@ Error YAMLProfileReader::readProfile(BinaryContext &BC) {
...
+  using BinaryFunctionProfileVector =
+      std::vector<llvm::yaml::bolt::BinaryFunctionProfile *,
+                  std::allocator<llvm::yaml::bolt::BinaryFunctionProfile *>>;
+
+  llvm::StringMap<BinaryFunctionProfileVector> LTOCommonNameMap;
```
Comment (maksfb) [inline bolt/lib/Profile/YAMLProfileReader.cpp:364]: This definition of `LTOCommonNameMap` will override the class member.

**Review by maksfb — CHANGES_REQUESTED** [review]

File: bolt/lib/Profile/YAMLProfileReader.cpp (hunk 1 above)
Comment (gz83) [reply bolt/lib/Profile/YAMLProfileReader.cpp:357]: It has been confirmed that the new code compiles properly on Windows.

File: bolt/lib/Profile/YAMLProfileReader.cpp (hunk 2 above)
Comment (gz83) [reply bolt/lib/Profile/YAMLProfileReader.cpp]: Done

**Comment by gz83** [comment]
I have modified the code as per your instructions, please review again, thank you!

@maksfb

**Review by maksfb — APPROVED** [review]
LGTM. Please add "Fix build issue on Windows" to the commit summary. Thanks for reporting and the fix.

**Comment by gz83** [comment]
> LGTM. Please add "Fix build issue on Windows" to the commit summary. Thanks for reporting and the fix.

Added them to the commit summary

@maksfb

**Comment by gz83** [comment]
3 checks passed

**Pull request merged**